.github/workflows/sui-protocol-version-monitor.yml

```diff
@@ -27,37 +27,50 @@ jobs:
     runs-on: [self-hosted, self-hosted-arc]
 
     steps:
-    - name: Read last protocol version in ${{ inputs.environment }}
-      uses: actions/download-artifact@v3
+    - name: Checkout sui source
+      uses: actions/checkout@7dd9e2a3dc350cf687eb1b2a4fadfee8c8e49675 # pin@v3
+
+    - name: Download an artifact for the last saved ${{ inputs.environment }} protocol version
+      id: download-artifact
+      uses: dawidd6/action-download-artifact@v2
       with:
+        github_token: ${{secrets.GITHUB_TOKEN}}
+        workflow: sui-protocol-version-monitor.yml
+        workflow_conclusion: success
         name: ${{ inputs.environment }}-protocol-version
-    - shell: bash
+        check_artifacts: true
+        path: /tmp/${{ inputs.environment }}_saved_protocol_version.txt
+        if_no_artifact_found: warn
+
+    - name: Read last saved protocol version for ${{ inputs.environment }}
+      continue-on-error: true
+      shell: bash
       run: |
-        echo "saved_protocol_version=`cat /tmp/${{ inputs.environment }}_saved_protocol_version.txt`" >> $GITHUB_ENV
+        echo "saved_protocol_version=`cat /tmp/${{ inputs.environment }}_saved_protocol_version.txt/${{ inputs.environment }}_saved_protocol_version.txt || ''`" >> $GITHUB_ENV
```

Comment:
This looks fishy? Why is the file inside a directory with the same name as itself?